.github/workflows/ci-model-regression.yml

```
@@ -0,0 +1,276 @@
+        with:
+          service_account_key: ${{ secrets.GKE_SA_RASA_CI_GPU }}
+          project_id: ${{ secrets.GKE_SA_RASA_CI_GPU_PROJECT }}
+
+      # Get the GKE credentials so we can deploy to the cluster
+      - run: |-
+          gcloud container clusters get-credentials "${{ secrets.GKE_GPU_CLUSTER }}" --zone "$GKE_ZONE"
+
+      - name: Deploy Github Runner
+        run: |-
+          kubectl apply -f runner_deployment.yaml
+          kubectl -n github-runner rollout status --timeout=15m deployment/github-runner-$GITHUB_RUN_ID
+
+  model_regression_test_gpu:
+    name: Model Regression Tests - GPU
+    needs:
+    - deploy_runner_gpu
+    - read_test_configuration
+    env:
+      LD_LIBRARY_PATH: "/usr/local/cuda/extras/CUPTI/lib64:/usr/local/cuda/lib64:/usr/local/nvidia/lib:/usr/local/nvidia/lib64"
+    runs-on: [self-hosted, gpu]
+    strategy:
+      max-parallel: 1
+      matrix: ${{fromJson(needs.read_test_configuration.outputs.matrix)}}
+    if: "contains(github.event.pull_request.labels.*.name, 'runner:gpu') && (github.event.pull_request.author_association == 'COLLABORATOR' || github.event.pull_request.author_association == 'MEMBER') && contains(github.event.pull_request.labels.*.name, 'status:model-regression-tests')"
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up Python 3.7 🐍
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.7
+
+      - name: Read Poetry Version 🔢
+        run: |
+          echo "::set-env name=POETRY_VERSION::$(scripts/poetry-version.sh)"
+        shell: bash
+
+      - name: Install poetry 🦄
+        uses: Gr1N/setup-poetry@v1
+        with:
+          poetry-version: ${{ env.POETRY_VERSION }}
+
+      - name: Load Poetry Cached Libraries ⬇
+        uses: actions/cache@v1
+        with:
+          path: ~/.cache/pypoetry/virtualenvs
+          key: ${{ runner.os }}-poetry-3.7-${{ hashFiles('**/poetry.lock') }}
+          restore-keys: ${{ runner.os }}-poetry-3.7
+
+      - name: Install Dependencies 📦
+        run: |
+          poetry install --extras full
+          make install
+          
+      - name: Validate that GPUs are working
+        run: |-
+          poetry run python -c 'from tensorflow.python.client import device_lib; print(device_lib.list_local_devices())' || true
+          
+      - name: Checkout dataset
+        uses: actions/checkout@v2
+        with:
+          repository: ${{ secrets.DATASET_REPOSITORY }}
+          token: ${{ secrets.ML_TEST_SA_PAT }}
+          path: 'dataset'
+          
+      - name: Run test
+        env:
+          TFHUB_CACHE_DIR: ~/.tfhub_cache/
+          OMP_NUM_THREADS: 1
+        run: |-
+          echo ${{ matrix.dataset }} ${{ matrix.config }}
+          poetry run rasa --version
+          poetry run rasa train nlu -u dataset/${{ matrix.dataset }} -c dataset/configs/${{ matrix.config }} --out models/${{ matrix.dataset }}/${{ matrix.config }}
```

Since we should be training only on train split of the dataset.
```suggestion
          poetry run rasa train nlu -u dataset/${{ matrix.dataset }}/train -c dataset/configs/${{ matrix.config }} --out models/${{ matrix.dataset }}/${{ matrix.config }}
```

---

👍